### ISSUE TITLE
Quagga: Add logging support

### ISSUE THREAD

**Issue by mimugmail** [opnsense/plugins#205]
I'm quite near replacing some Quagga / Linux units with OPNsense, but I'm missing a log.
Yesterday I switched some BGP stuff on my live systems and watched the log:

2017/07/18 22:08:32 BGP: %NOTIFICATION: sent to neighbor 1.1.1.4 6/6 (Cease/Other Configuration Change) 0 bytes
2017/07/18 22:08:32 BGP: Notification sent to neighbor 1.1.1.4: configuration change
2017/07/18 22:08:32 BGP: %NOTIFICATION: sent to neighbor 1.1.1.26 6/6 (Cease/Other Configuration Change) 0 bytes
2017/07/18 22:08:32 BGP: Notification sent to neighbor 1.1.1.26: configuration change
2017/07/18 22:08:32 BGP: %NOTIFICATION: sent to neighbor 1.1.1.46 6/6 (Cease/Other Configuration Change) 0 bytes
2017/07/18 22:08:32 BGP: Notification sent to neighbor 1.1.1.46: configuration change
2017/07/18 22:08:32 BGP: %NOTIFICATION: sent to neighbor 1.1.1.47 6/6 (Cease/Other Configuration Change) 0 bytes
2017/07/18 22:08:32 BGP: Notification sent to neighbor 1.1.1.47: configuration change
2017/07/18 22:08:32 BGP: %ADJCHANGE: neighbor 1.1.1.4 Down Peer-group add member
2017/07/18 22:08:32 BGP: %ADJCHANGE: neighbor 1.1.1.26 Down Peer-group add member
2017/07/18 22:08:32 BGP: %ADJCHANGE: neighbor 1.1.1.46 Down Peer-group add member
2017/07/18 22:08:32 BGP: %ADJCHANGE: neighbor 1.1.1.47 Down Peer-group add member
2017/07/18 22:08:36 BGP: stream_read_try: read failed on fd 15: Connection reset by peer
2017/07/18 22:08:37 BGP: stream_read_try: read failed on fd 14: Connection reset by peer
2017/07/18 22:08:37 BGP: 1.1.1.46 [Error] bgp_read_packet error: Connection reset by peer
2017/07/18 22:08:39 BGP: stream_read_try: read failed on fd 15: Connection reset by peer
2017/07/18 22:08:39 BGP: 1.1.1.47 [Error] bgp_read_packet error: Connection reset by peer
2017/07/18 22:08:42 BGP: %ADJCHANGE: neighbor 1.1.1.46 Up
2017/07/18 22:08:45 BGP: %ADJCHANGE: neighbor 1.1.1.47 Up
2017/07/18 22:08:46 BGP: %ADJCHANGE: neighbor 1.1.1.26 Up
2017/07/18 22:08:48 BGP: %ADJCHANGE: neighbor 1.1.1.4 Up

We should also have some kind of loggin for this.
With quagga you can log to syslog of a plain file. 

I'd suggest filelogging with:
/var/log/quagga/ 
and one file for each daemon:
bgpd.log, ospfd.log etc.

Opinions @fabianfrz @fichtner ?

**Comment by fabianfrz** [comment]
I would prefer syslog as well as the data can go to an ELK stack as well. A single log should do it as well because if you look at your sample it contains the routing protocol.

**Comment by mimugmail** [comment]
Ok, there's already a config in /usr/local/etc/inc/system.inc

`        $syslogconfs['routing'] = array('facility' => array('radvd', 'routed', 'rtsold', 'olsrd', 'zebra', 'ospfd', 'bgpd', 'miniupnpd'));
`

So I think it's just fine to put in all configs:

log syslog informational

Or is this too aggressive @fichtner ?

**Comment by fichtner** [comment]
create quagga files? or quagga-"PROTOCOL" ?

**Comment by mimugmail** [comment]
@fichtner in zebra.conf, ospfd.conf and bgpd.conf just a

`log syslog informational`

It looks like this in routing.log:

```
Jul 21 16:09:30 OPNsense bgpd[69132]: Terminating on signal
Jul 21 16:09:30 OPNsense bgpd[69132]: VRF 0 is to be deleted.
Jul 21 16:09:30 OPNsense bgpd[69132]: VRF 0 is to be disabled.
Jul 21 16:09:30 OPNsense zebra[67619]: client 12 disconnected. 0 bgp routes removed from the rib
Jul 21 16:09:30 OPNsense zebra[67619]: Terminating on signal
Jul 21 16:09:30 OPNsense zebra[75077]: Zebra 1.2.1 starting: vty@2601
Jul 21 16:09:30 OPNsense bgpd[77132]: BGPd 1.2.1 starting: vty@2605, bgp@<all>:179 pid 77132
Jul 21 16:09:30 OPNsense zebra[75077]: client 12 says hello and bids fair to announce only bgp routes
Jul 21 16:09:30 OPNsense zebra[75077]: Try to set TE Link Param
Jul 21 16:09:30 OPNsense zebra[75077]: Try to set TE Link Param
Jul 21 16:09:30 OPNsense zebra[75077]: Try to set TE Link Param
Jul 21 16:09:30 OPNsense zebra[75077]: Try to set TE Link Param
Jul 21 16:09:30 OPNsense zebra[75077]: Try to set TE Link Param
Jul 21 16:09:30 OPNsense zebra[75077]: Try to set TE Link Param
Jul 21 16:09:31 OPNsense bgpd[77132]: can't connect to 81.24.66.163 fd 12 : Host is down
Jul 21 16:09:36 OPNsense bgpd[77132]: can't connect to 81.24.66.163 fd 12 : Host is down
Jul 21 16:09:41 OPNsense bgpd[77132]: can't connect to 81.24.66.163 fd 12 : Host is down

```

**Comment by fichtner** [comment]
we need to have log files for plugins, it's very hard to check a "routing" log, it's already too messed up with miniupnpd, radvd, etc.

**Comment by mimugmail** [comment]
So create a /var/log/quagga.log and point the processes to it? I'm not quite familiar with the logic in the inc file :)

**Comment by fichtner** [comment]
https://github.com/opnsense/plugins/blob/master/security/tinc/src/etc/inc/plugins.inc.d/tinc.inc#L42-L59

**Comment by fabianfrz** [comment]
@mimugmail I've added the feature in #206, it does not look like we have to update the file.

**Comment by fabianfrz** [comment]
fixed by #206
